gibbon-compiler/src/Gibbon/L2/Syntax.hs

```diff
@@ -478,14 +479,36 @@ data Region = GlobR Var Multiplicity -- ^ A global region with lifetime equal to
                                      --   are no free locations in the program.
   deriving (Read,Show,Eq,Ord, Generic)
 
+data ExtendedRegion = AoSR Region                           -- ^ A simple "flat" region where the datatype 
+                                                            --   will reside in an array of structure representation.
+                    | SoAR Region [((String, Int), Region)] -- ^ A complex region representation for a datatype 
```

Comment:
Any chance for a named type instead of `(String, Int)`? It's really hard to grok what's going on with tuples of builtin types like Int and String.

Reply:
sure